src/Pgvector/Vector.cs

```diff
@@ -4,22 +4,60 @@
 
 namespace Pgvector;
 
-public class Vector
+public class Vector : IEquatable<Vector>
 {
-    private ReadOnlyMemory<float> vec;
+    public ReadOnlyMemory<float> Memory { get; }
 
     public Vector(ReadOnlyMemory<float> v)
-        => vec = v;
-
-    public Vector(float[] v)
```

Comment:
Note that arrays are implicitly convertible to Memory/ReadOnlyMemory, so there's no need for this constructor.